src/gui/mzroll/mainwindow.cpp

```diff
@@ -1985,56 +1993,83 @@ bool MainWindow::loadMetaInformation(QString filename) {
 	}
 }
 
-void MainWindow::loadCompoundsFile() {
-	QStringList filelist =
-			QFileDialog::getOpenFileNames(this, "Select Compounds File To Load",
-					".",
-					"All Known Formats(*.csv *.tab *.tab.txt *.msp *.sptxt *.pepXML *.massbank);;Tab Delimited(*.tab);;Tab Delimited Text(*.tab.txt);;CSV File(*.csv);;NIST Library(*.msp);;SpectraST(*.sptxt);;pepXML(*.pepXML);;MassBank(*.massbank");
-
-    if ( filelist.size() == 0 || filelist[0].isEmpty() ) return;
-	if(!loadCompoundsFile(filelist[0])) {
-		string dbfilename = filelist[0].toStdString();
-		string dbname = mzUtils::cleanFilename(dbfilename);
-		string notFoundColumns = "Following are the unknown column name(s) found: ";
+void MainWindow::loadCompoundsFile()
+{
+    QStringList filelist = QFileDialog::getOpenFileNames(
+        this,
+        "Select Compounds File To Load",
+        ".",
+        "All Known Formats(*.csv *.tab *.tab.txt *.msp *.sptxt *.pepXML "
+        "*.massbank);;Tab Delimited(*.tab);;Tab Delimited Text(*.tab.txt);;CSV "
+        "File(*.csv);;NIST "
+        "Library(*.msp);;SpectraST(*.sptxt);;pepXML(*.pepXML);;MassBank(*."
+        "massbank");
+
+    if (filelist.size() == 0 || filelist[0].isEmpty())
+        return;
+    if (!loadCompoundsFile(filelist[0])) {
+        string dbfilename = filelist[0].toStdString();
+        string dbname = mzUtils::cleanFilename(dbfilename);
+        string notFoundColumns = "Following are the unknown column name(s) "
+                                 "found: ";
 
         QMessageBox msgBox;
-		msgBox.setText(tr("Trouble in loading compound database %1").arg(QString::fromStdString(dbname)));
+        msgBox.setText(tr("Trouble in loading compound database %1")
+                         .arg(QString::fromStdString(dbname)));
         msgBox.setIcon(QMessageBox::Warning);
-		if (DB.notFoundColumns.size() > 0) {
-			for(std::vector<string>::iterator it = DB.notFoundColumns.begin(); it != DB.notFoundColumns.end(); ++it) {
-    			notFoundColumns += "\n" + *it;
-			}
-			msgBox.setDetailedText(QString::fromStdString(notFoundColumns));
-			msgBox.setWindowFlags(msgBox.windowFlags() & ~Qt::WindowCloseButtonHint);
-		}
+        if (DB.notFoundColumns.size() > 0) {
+            for (std::vector<string>::iterator it = DB.notFoundColumns.begin();
+                 it != DB.notFoundColumns.end();
+                 ++it) {
+                notFoundColumns += "\n" + *it;
+            }
+            msgBox.setDetailedText(QString::fromStdString(notFoundColumns));
+            msgBox.setWindowFlags(msgBox.windowFlags()
+                                  & ~Qt::WindowCloseButtonHint);
+        }
+        analytics->hitEvent("Load Compound DB",
+                            "Column Error",
+                            DB.notFoundColumns.size());
```

Comment:
the number of unknown columns might not be of use when analysing this data. Maybe the value can be whether the db was successfully loaded or not. The `if` block is for successfull load but some additional/unknown columns were found. The `else` block is for error in upload because of missing/unknown columns.

Reply:
Yeah, row error does not need a value.